cortex-mixin/recording_rules.libsonnet

```
@@ -69,12 +69,21 @@ local utils = import 'mixin-utils/utils.libsonnet';
         rules: [
           {
             // Convenience rule to get the number of replicas for both a deployment and a statefulset.
+            // Multi-zone deployments are grouped together removing the "zone-X" suffix.
             record: 'cluster_namespace_deployment:actual_replicas:count',
             expr: |||
-              sum by (cluster, namespace, deployment) (kube_deployment_spec_replicas)
-                or
               sum by (cluster, namespace, deployment) (
-                label_replace(kube_statefulset_replicas, "deployment", "$1", "statefulset", "(.*)")
+                label_replace(
+                  kube_deployment_spec_replicas,
+                  "deployment", "$1", "deployment", "(.*?)(?:-zone-[a-z])?"
+                )
+              )
+              or
+              sum by (cluster, namespace, deployment) (
+                label_replace(
+                  label_replace(kube_statefulset_replicas, "deployment", "$1", "statefulset", "(.*)"),
+                  "deployment", "$1", "deployment", "(.*?)(?:-zone-[a-z])?"
+                )
```

The inner label replace is just moving the `statefuleset` label to the deployment label, so could be done with this I believe:

```suggestion
                label_replace(kube_statefulset_replicas, "deployment", "$1", "statefulset", "(.*?)(?:-zone-[a-z])?"),
```

---

That's right. I've applied the suggested change and manually tested it.